_extensions/flourish/flourish.js

```diff
@@ -45,14 +49,27 @@ document.addEventListener("DOMContentLoaded", () => {
                 }
 
                 // Apply flourishes with the appropriate class name
-                content = injectFlourishes(content, new RegExp(pattern.regex.source, 'g'), className, pattern.mask);
+                content = injectFlourishes(content, pattern.regex, className, pattern.mask);
             }
 
             el.innerHTML = content;
+            el.dataset.flourished = 'true';
         }
     }
 });
 
+// Hardcode swaps for sanitized HTML
+function htmlEntityEscapeForSearch(s) {
+    return s
+        .replace(/</g, '&lt;')
+        .replace(/>/g, '&gt;')
+        .replace(/&(?![a-zA-Z]+;|#\d+;|#x[0-9A-Fa-f]+;)/g, '&amp;');
```

Comment:
I think if we just do this one first, we shoudn't need the fancy look-ahead to make sure we don't replace &lt; etc.